contrib/msggen/msggen/schema.json

```diff
@@ -16136,6 +16136,168 @@
         }
       ]
     },
+    "lightning-listaddresses.json": {
+      "$schema": "../rpc-schema-draft.json",
+      "type": "object",
+      "added": "v24.11",
+      "rpc": "listaddresses",
+      "title": "Command to list all addresses issued by the node to date",
+      "description": [
+        "The **listaddresses** RPC command provides a detailed list of all Bitcoin addresses that have been generated and issued by the Core Lightning node up to the current date."
+      ],
+      "categories": [
+        "readonly"
+      ],
+      "request": {
+        "required": [],
+        "additionalProperties": false,
+        "properties": {
+          "start": {
+            "type": "u64",
+            "description": [
+              "Starting index for listing addresses or searching for a particular address."
+            ],
+            "default": 0
+          },
+          "end": {
```

Comment:
It's customary to implement paging as an offset + limit, not as absolute indices, otherwise you need to handle strange cases like start > end, and the issue below with future addresses (which should not be exposed imho).

Reply:
Thanks, I should have noticed it. Fixed now!